Geometry/HGCalCommonData/src/HGCalCellUV.cc

```diff
@@ -0,0 +1,275 @@
+          cellPosFine_[placement][std::pair<int, int>(u, v)] = hgcalcell.HGCalCellUV2XY1(u, v, placement, 0);
+        }
+      }
+    }
+    // Coarse cells
+    for (int iu = 0; iu < 2 * ncell_[1]; ++iu) {
+      for (int iv = 0; iv < 2 * ncell_[1]; ++iv) {
+        int u = (placement < HGCalCell::cellPlacementExtra) ? iv : iu;
+        int v = (placement < HGCalCell::cellPlacementExtra) ? iu : iv;
+        if (((v - u) < ncell_[1]) && (u - v) <= ncell_[1]) {
+          cellPosCoarse_[placement][std::pair<int, int>(u, v)] = hgcalcell.HGCalCellUV2XY1(u, v, placement, 1);
+        }
+      }
+    }
+  }
+}
+
+std::pair<int32_t, int32_t> HGCalCellUV::HGCalCellUVFromXY0(
+    double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug) {
+  //--- Reverse transform to placement=0, if placement index ≠ 6
+  double xloc1 = (placement >= 6) ? xloc : -xloc;
+  int rot = placement % 6;
+  const std::vector<double> fcos = {1.0, 0.5, -0.5, -1.0, -0.5, 0.5};
+  const std::vector<double> fsin = {0.0, sqrt3By2_, sqrt3By2_, 0.0, -sqrt3By2_, -sqrt3By2_};
+  double xprime = xloc1 * fcos[rot] - yloc * fsin[rot];
+  double yprime = xloc1 * fsin[rot] + yloc * fcos[rot];
+  double x = xprime;
+  double y = yprime;
+
+  //--- Calculate coordinates in u,v,w system
+  const double sin60(sqrt3By2_), cos60(0.5);
+  double u = x * sin60 + y * cos60;
+  double v = -x * sin60 + y * cos60;
+  double w = y;
+
+  //--- Rounding in u, v, w coordinates
+  int iu = std::floor(u / cellY_[type]) + 3 * (ncell_[type]) - 1;
+  int iv = std::floor(v / cellY_[type]) + 3 * (ncell_[type]);
+  int iw = std::floor(w / cellY_[type]) + 1;
+
+  int isv = (iu + iw) / 3;
+  int isu = (iv + iw) / 3;
+
+  //--- Taking care of extending cells
+  if ((iu + iw) < 0) {
+    isu = (iv + iw + 1) / 3;
+    isv = 0;
+  } else if (isv - isu > ncell_[type] - 1) {
+    isu = (iv + iw + 1) / 3;
+    isv = (iu + iw - 1) / 3;
+  } else if (isu > 2 * ncell_[type] - 1) {
+    isu = 2 * ncell_[type] - 1;
+    isv = (iu + iw - 1) / 3;
+  }
+  if (debug)
+    edm::LogVerbatim("HGCalGeom") << "HGCalCellUVFromXY0: Input " << xloc << ":" << yloc << ":" << extend << " Output "
+                                  << isu << ":" << isv;
+  return std::make_pair(isu, isv);
+}
+
+std::pair<int32_t, int32_t> HGCalCellUV::HGCalCellUVFromXY1(
+    double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug) {
+  //--- Using multiple inequalities to find (u, v)
+  //--- Reverse transform to placement=0, if placement index ≠ 7
+  double x = (placement >= 6) ? xloc : -1 * xloc;
+  const std::vector<double> fcos = {0.5, 1.0, 0.5, -0.5, -1.0, -0.5};
+  const std::vector<double> fsin = {-sqrt3By2_, 0.0, sqrt3By2_, sqrt3By2_, 0.0, -sqrt3By2_};
+  double y = x * fsin[placement % 6] + yloc * fcos[placement % 6];
+  x = x * fcos[placement % 6] - yloc * fsin[placement % 6];
+
+  int32_t u(-100), v(-100);
+  int N_ = (type != 0) ? ncell_[1] : ncell_[0];
```

Comment:
`N_` is not an allowed variable name. Please use a descriptive variable name that starts with lowercase.